wezterm-gui/src/termwindow/charselect.rs

```diff
@@ -508,32 +508,54 @@ impl CharSelector {
         *self.top_row.borrow_mut() = 0;
     }
 
-    fn move_up(&self) {
-        let mut row = self.selected_row.borrow_mut();
-        *row = row.saturating_sub(1);
-
-        let mut top_row = self.top_row.borrow_mut();
-        if *row < *top_row {
-            *top_row = *row;
+    fn move_up(&self, count: usize) {
```

Comment:
Encoding special logic for `0` or `-1` is usually a good sign that it's time for an enum!

I'd suggest something like:

```rust
enum Move {
   Up(usize),
   Down(usize),
   PageUp,
   PageDown,
}
```

Then you can roll `move_up` and `move_down` into a single function:

```rust
fn move(&self, how: Move) {
   let page_size = *self.max_rows_on_screen.borrow();
   let current_row = *self.selected_row.borrow();
   let dest = match how {
       Move::Up(n) => current_row.saturating_sub(n),
       Move::PageUp => current_row.saturating_sub(page_size),
       Move::Down(n) => current_row.saturating_add(n),
       Move::PageDown => current_row.saturating_add(page_size),
   };
   *self.selected_row.borrow_mut() = dest;
   self.nav_selection();
}
```

Reply:
lol, thats what i wanted to avoid... for now, also because i would probably roll up a few of your list/row selection gimmicks with the same enum/trait/interface.... thats how i at least handle stuff in Go... a single 

```go
func nav_selection(i int)*Object{}
```

that is used with +/- depending on direction... 

```
func moveUp(){nav_selection(-1)}
func moveDown(){nav_selection(1)}
```